src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/persistence/repository/MigrationHistoryRepository.kt

```diff
@@ -1,7 +1,29 @@
 package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.persistence.repository
 
+import kotlinx.coroutines.flow.Flow
+import org.springframework.data.r2dbc.repository.Query
 import org.springframework.data.repository.kotlin.CoroutineCrudRepository
+import org.springframework.data.repository.query.Param
 import org.springframework.stereotype.Repository
+import java.time.LocalDateTime
 
 @Repository
-interface MigrationHistoryRepository : CoroutineCrudRepository<MigrationHistory, String>
```

Comment:
Non of this is pretty, and see the hack in the service that is worse; I might give https://docs.spring.io/spring-data/r2dbc/docs/current/api/org/springframework/data/r2dbc/core/R2dbcEntityTemplate.html a go and see if it works better